### ISSUE TITLE
Trouble with downloading releases

### ISSUE THREAD

**Issue by mondkaefer** [jborean93/ansible-role-win_openssh#5]
Hi Jordan,

First: nice work, I like this role.

Ansible version: 2.9.1
Two issues I run into, both related to redirects.

1. Download latest release. 
Task "get latest release version" in download.yml fails. Error message with follow_redirects set to safe: "Status code of request '302' is not in list of valid status codes 200 : Redirect'."

Full log: 
```
TASK [jborean93.win_openssh : get latest release version] ****
fatal: [<SNIPPED IP_ADDRESS>]: FAILED! => {
	"cache_control": "no-cache",
	"changed": false,
	"character_set": "utf-8",
	"content_encoding": "",
	"content_length": -1,
	"content_security_policy": "default-src 'none'; base-uri 'self'; block-all-mixed-content; connect-src 'self' uploads.github.com www.githubstatus.com collector.githubapp.com api.github.com www.google-analytics.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com wss://live.github.com; font-src github.githubassets.com; form-action 'self' github.com gist.github.com; frame-ancestors 'none'; frame-src render.githubusercontent.com; img-src 'self' data: github.githubassets.com identicons.github.com collector.githubapp.com github-cloud.s3.amazonaws.com *.githubusercontent.com; manifest-src 'self'; media-src 'none'; script-src github.githubassets.com; style-src 'unsafe-inline' github.githubassets.com",
	"content_type": "text/html; charset=utf-8",
	"cookies": [],
	"date": "Fri, 06 Dec 2019 00:22:37 GMT",
	"elapsed": 0,
	"expect_ct": "max-age=2592000, report-uri=\"https://api.github.com/_private/browser/errors\"",
	"headers": ["Status", "Vary", "Cache-Control", "Content-Type", "Date", "Location", "Set-Cookie", "Server", "X-Request-Id", "Strict-Transport-Security", "X-Frame-Options", "X-Content-Type-Options", "X-XSS-Protection", "Expect-CT", "Content-Security-Policy", "X-GitHub-Request-Id"],
	"is_from_cache": false,
	"is_mutually_authenticated": false,
	"last_modified": "2019-12-06T00:22:36.5829186+13:00",
	"location": "https://github.com/PowerShell/Win32-OpenSSH/releases/tag/v8.0.0.0p1-Beta",
	"method": "HEAD",
	"msg": "Status code of request '302' is not in list of valid status codes 200 : Redirect'.",
	"protocol_version": {
		"Build": -1,
		"Major": 1,
		"MajorRevision": -1,
		"Minor": 1,
		"MinorRevision": -1,
		"Revision": -1
	},
	"response_uri": "https://github.com/PowerShell/Win32-OpenSSH/releases/latest",
	"server": "GitHub.com",
	"set_cookie": "has_recent_activity=1; path=/; expires=Fri, 06 Dec 2019 01:22:37 -0000,logged_in=no; domain=.github.com; path=/; expires=Tue, 06 Dec 2039 00:22:37 -0000; secure; HttpOnly,_gh_sess=aGYraFNFc2dwdy9JTkk0cnc2M1kvemVmdU1RZ01sSFdYU0I3dXFXQ2pzMjg0WHA3RG1oVGZlODl0ck1HK09ITCtWRlR3WnIxUTluQk1ndmc2WmZseDJaOEJMRVZLVVQrN2c3VXA2cUxWVC9WaTIxTlNpQldWRlk5STNvTHlDcjZFRzAwQUlZQmR3aVQ0OE5HWHhvOUlJMmJEazBBSmsyTDF0V3d4RURucU5ZPS0tRWc4b1dPRWFkNWdQZDd6R1ZWSmdnZz09--826120a850000753d65339b0165a5c8f476a1068; path=/; secure; HttpOnly",
	"status": "302 Found",
	"status_code": 302,
	"status_description": "Found",
	"strict_transport_security": "max-age=31536000; includeSubdomains; preload",
	"supports_headers": true,
	"url": "https://github.com/PowerShell/Win32-OpenSSH/releases/latest",
	"vary": "X-PJAX,Accept-Encoding",
	"x_content_type_options": "nosniff",
	"x_frame_options": "deny",
	"x_git_hub_request_id": "9DC0:775C:849CA2:CF002B:5DE99F4C",
	"x_request_id": "4c12914a-c599-4bf2-af46-3cf1fa80c27c",
	"xxss_protection": "1; mode=block"
}

```
Changing the value of follow_redirects from safe to all fixes this.

2. If I specify a release, say v8.0.0.0p1-Beta, it fails in task "extract the Win32-OpenSSH zip to a temp folder" with an unzip error because the file downloaded is not a valid zip file, but an html file with a note that we're being redirected.

Full log: 
```
TASK [jborean93.win_openssh : extract the Win32-OpenSSH zip to a temp folder] ********
fatal: [<SNIPPED IP_ADDRESS>]: FAILED! => {
	"changed": false,
	"dest": "C:\\Windows\\TEMP\\OpenSSH",
	"msg": "Error unzipping 'C:\\Windows\\TEMP\\OpenSSH-Win.zip' to 'C:\\Windows\\TEMP\\OpenSSH'!. Method: System.IO.Compression.ZipFile, Exception: Exception calling \"Open\" with \"3\" argument(s): \"End of Central Directory record could not be found.\"",
	"removed": false,
	"src": "C:\\Windows\\TEMP\\OpenSSH-Win.zip"
}
```

Adding follow_redirects with value all fixes this.

Happy to provide a PR if you'd like.

Cheers,
Martin

**Comment by jborean93** [comment]
Thanks for the report, I think this is the same issue that has been reported at https://github.com/ansible/ansible/issues/65556. I've found the problem and it should be fixed in the next Ansible release.

**Comment by mondkaefer** [comment]
That would make sense. Being fairly new to ansible I thought that follow_redirects: safe should be sufficient - provided it works correctly. Thanks for the quick follow-up.

**Comment by jborean93** [comment]
Closing as this is a bug in Ansible itself and will be fixed in the next Ansible 2.9 release. Thanks for reporting this.
